contracts/BondDepository.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "./Factory.sol";
+import "./Governable.sol";
+import "./interface/ISOLACE.sol";
+import "./interface/IxSOLACE.sol";
+import "./interface/IBondTeller.sol";
+import "./interface/IBondDepository.sol";
+
+/**
+ * @title BondDepository
+ * @author solace.fi
+ * @notice Factory for [`bond tellers`](./BondTeller).
+ */
+contract BondDepository is IBondDepository, Factory, Governable {
+
+    // pass these when initializing tellers
+    ISOLACE internal _solace;
+    IxSOLACE internal _xsolace;
+    address internal _pool;
+    address internal _dao;
+
+    // track tellers
+    mapping(address => bool) internal _isTeller;
+
+    /**
+     * @notice Constructs the BondDepository contract.
+     * @param governance The address of the [governor](/docs/protocol/governance).
+     * @param solace Address of [**SOLACE**](./solace).
+     * @param solace Address of [**xSOLACE**](./xsolace).
+     * @param pool Address of [`UnderwritingPool`](./underwritingpool).
+     * @param solace Address of the DAO.
+     */
+    constructor(address governance, address solace, address xsolace, address pool, address dao) Governable(governance) {
```

Comment:
I think we can provide solace, xsolace, pool and dao via registry. But it is not big deal.

Reply:
registry doesnt support those types. v2 should allow for arbitrary adds and lookups